File: backend/internal/models/anime.go
package models

// AnimeFilter enthält alle Filterparameter für Anime-Listenabfragen in der öffentlichen und Admin-API.
type AnimeFilter struct {
	Page            int    // Seitennummer (1-basiert)
	PerPage         int    // Einträge pro Seite
	Letter          string // Anfangsbuchstabe des Titels
	Q               string // Volltextsuchbegriff
	ContentType     string // Inhaltstyp-Filter (z.B. "anime", "movie")
	Status          string // Status-Filter (z.B. "ongoing", "completed")
	FansubGroupID   *int64 // Optional: nur Anime einer bestimmten Fansub-Gruppe
	HasCover        *bool  // Optional: nur Anime mit oder ohne Coverbild
	IncludeDisabled bool   // Deaktivierte Einträge einschließen (nur Admin)
}

// AnimeListItem repräsentiert einen einzelnen Anime-Eintrag in der öffentlichen Listenansicht.
type AnimeListItem struct {
	ID          int64   `json:"id"`
	Title       string  `json:"title"`
	Type        string  `json:"type"`
	ContentType string  `json:"content_type,omitempty"`
	Status      string  `json:"status"`
	Year        *int16  `json:"year,omitempty"`
	CoverImage  *string `json:"cover_image,omitempty"`
	MaxEpisodes *int16  `json:"max_episodes,omitempty"`
}

// EpisodeListItem repräsentiert eine einzelne Episode in der Episodenliste eines Anime.
type EpisodeListItem struct {
	ID            int64    `json:"id"`
	EpisodeNumber string   `json:"episode_number"`
	Title         *string  `json:"title,omitempty"`
	Status        string   `json:"status"`
	ViewCount     int32    `json:"view_count"`
	DownloadCount int32    `json:"download_count"`
	StreamLinks   []string `json:"stream_links,omitempty"`
	Filename      *string  `json:"filename,omitempty"`
}

// AnimeDetail enthält die vollständige Detailansicht eines Anime inklusive Episodenliste,
// wird von der öffentlichen Anime-Detailseite verwendet.
type AnimeDetail struct {
	ID          int64             `json:"id"`
	Title       string            `json:"title"`
	TitleDE     *string           `json:"title_de,omitempty"`
	TitleEN     *string           `json:"title_en,omitempty"`
	Type        string            `json:"type"`
	ContentType string            `json:"content_type"`
	Status      string            `json:"status"`
	Year        *int16            `json:"year,omitempty"`
	MaxEpisodes *int16            `json:"max_episodes,omitempty"`
	Genre       *string           `json:"genre,omitempty"`
	Genres      []string          `json:"genres,omitempty"`
	Description *string           `json:"description,omitempty"`
	CoverImage  *string           `json:"cover_image,omitempty"`
	ViewCount   int32             `json:"view_count"`
	Episodes    []EpisodeListItem `json:"episodes"`
}

// AnimeMediaLookup enthält die für Medienpfad-Lookups benötigten Titelfelder eines Anime,
// wird intern für Jellyfin- und Medien-Resolver-Abfragen verwendet.
type AnimeMediaLookup struct {
	Title      string
	TitleDE    *string
	TitleEN    *string
	Source     *string
	FolderName *string
}

// AnimeBackdropManifest enthält alle Backdrop- und Medien-URLs eines Anime,
// die von einem externen Provider (z.B. Jellyfin) geliefert werden.
type AnimeBackdropManifest struct {
	AnimeID     int64    `json:"anime_id"`
	Provider    string   `json:"provider"`
	MediaItemID string   `json:"media_item_id,omitempty"`
	Backdrops   []string `json:"backdrops"`
	ThemeVideos []string `json:"theme_videos"`
	LogoURL     string   `json:"logo_url,omitempty"`
	BannerURL   string   `json:"banner_url,omitempty"`
}

// PaginationMeta enthält die Paginierungsmetadaten für paginierten Listenendpunkte.
type PaginationMeta struct {
	Total      int64 `json:"total"`
	Page       int   `json:"page"`
	PerPage    int   `json:"per_page"`
	TotalPages int   `json:"total_pages"`
}

// GenreToken repräsentiert ein Genre mit seiner Verwendungshäufigkeit im Anime-Katalog.
type GenreToken struct {
	Name  string `json:"name"`
	Count int64  `json:"count"`
}

File: frontend/src/types/anime.ts
export type AnimeStatus = 'disabled' | 'ongoing' | 'done' | 'aborted' | 'licensed'
export type ContentType = 'anime' | 'hentai'

export interface AnimeListItem {
  id: number
  title: string
  type: string
  status: AnimeStatus
  year?: number
  cover_image?: string
  max_episodes?: number
}

export interface PaginationMeta {
  total: number
  page: number
  per_page: number
  total_pages: number
}

export interface PaginatedAnimeResponse {
  data: AnimeListItem[]
  meta: PaginationMeta
}

export interface AnimeListParams {
  page?: number
  per_page?: number
  q?: string
  letter?: string
  content_type?: ContentType
  status?: AnimeStatus
  fansub_id?: number
  has_cover?: boolean
  include_disabled?: boolean
}

export type EpisodeStatus = 'disabled' | 'private' | 'public'

export interface EpisodeListItem {
  id: number
  episode_number: string
  title?: string
  status: EpisodeStatus
  view_count: number
  download_count: number
  stream_links?: string[]
}

export interface AnimeDetail {
  id: number
  title: string
  title_de?: string
  title_en?: string
  type: string
  content_type: ContentType
  status: AnimeStatus
  year?: number
  max_episodes?: number
  genre?: string
  genres?: string[]
  tags?: string[]
  description?: string
  cover_image?: string
  source?: string
  folder_name?: string
  jellyfin_series_id?: string
  jellyfin_series_path?: string
  view_count: number
  episodes: EpisodeListItem[]
}

export interface EpisodeDetail {
  id: number
  anime_id: number
  anime_title: string
  episode_number: string
  title?: string
  status: EpisodeStatus
  view_count: number
  download_count: number
  stream_links?: string[]
  previous_episode_id?: number
  next_episode_id?: number
}

export interface AnimeBackdropManifest {
  anime_id: number
  provider: string
  media_item_id?: string
  theme_videos: string[]
  backdrops: string[]
  banner_url?: string
  logo_url?: string
}

export interface AnimeBackdropResponse {
  data: AnimeBackdropManifest
}

export interface AnimeRelation {
  anime_id: number
  title: string
  relation_type: string
  cover_image: string | null
  year: number | null
  type: string
}

export interface AnimeRelationsResponse {
  data: AnimeRelation[]
}
